editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ImageProvider.java

```diff
@@ -209,10 +206,6 @@ public static String getKeyForLargeIcon(String iconName) {
 	}
 	
 	protected static boolean isImageAvailable(String iconName) {
-		if (CamelUtils.getDiagramEditor() != null) {
-			Image img = GraphitiUi.getImageService().getImageForId(CamelUtils.getDiagramEditor().getDiagramTypeProvider().getProviderId(), iconName);
-			return img != null;
-		}
```

Comment:
(put here our direct discussion for records)
It is the cause of the issue.
in the code, there was a line searching in graphiti image provider if an image exists for a specific "iconName". the first time it is called, it returns null. The next time it returns a "red square" Image and so it doesn't return anymore null
it was causing the red squares in palette when doing a search